### PR TITLE
HBASE-22622 - WALKey Extended Attributes

### PR DESCRIPTION
**Pull request by gjacoby126**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 24 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| 0 | mvndep | 25 | Maven dependency ordering for branch |
| +1 | mvninstall | 255 | master passed |
| +1 | compile | 114 | master passed |
| +1 | checkstyle | 87 | master passed |
| +1 | shadedjars | 264 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 439 | master passed |
| +1 | javadoc | 60 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 16 | Maven dependency ordering for patch |
| +1 | mvninstall | 241 | the patch passed |
| +1 | compile | 110 | the patch passed |
| +1 | cc | 110 | the patch passed |
| +1 | javac | 110 | the patch passed |
| -1 | checkstyle | 68 | hbase-server: The patch generated 10 new + 14 unchanged - 0 fixed = 24 total (was 14) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 268 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 730 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | hbaseprotoc | 109 | the patch passed |
| +1 | findbugs | 472 | the patch passed |
| +1 | javadoc | 57 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 38 | hbase-protocol-shaded in the patch passed. |
| +1 | unit | 26 | hbase-protocol in the patch passed. |
| +1 | unit | 8200 | hbase-server in the patch passed. |
| +1 | asflicense | 84 | The patch does not generate ASF License warnings. |
| | | 12059 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-352/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/352 |
| Optional Tests |  dupname  asflicense  cc  unit  hbaseprotoc  javac  javadoc  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 0d338d0d00ee 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / b22459c37c |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-352/1/artifact/out/diff-checkstyle-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-352/1/testReport/ |
| Max. process+thread count | 4944 (vs. ulimit of 10000) |
| modules | C: hbase-protocol-shaded hbase-protocol hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-352/1/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by gjacoby126** [comment]
Pushed fixes for checkstyle nits.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 52 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| 0 | mvndep | 27 | Maven dependency ordering for branch |
| +1 | mvninstall | 252 | master passed |
| +1 | compile | 110 | master passed |
| +1 | checkstyle | 86 | master passed |
| +1 | shadedjars | 268 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 444 | master passed |
| +1 | javadoc | 59 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 14 | Maven dependency ordering for patch |
| +1 | mvninstall | 236 | the patch passed |
| +1 | compile | 111 | the patch passed |
| +1 | cc | 111 | the patch passed |
| +1 | javac | 111 | the patch passed |
| -1 | checkstyle | 69 | hbase-server: The patch generated 1 new + 14 unchanged - 0 fixed = 15 total (was 14) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 266 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 732 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | hbaseprotoc | 108 | the patch passed |
| +1 | findbugs | 453 | the patch passed |
| +1 | javadoc | 58 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 35 | hbase-protocol-shaded in the patch passed. |
| +1 | unit | 24 | hbase-protocol in the patch passed. |
| -1 | unit | 15297 | hbase-server in the patch failed. |
| +1 | asflicense | 82 | The patch does not generate ASF License warnings. |
| | | 19160 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-352/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/352 |
| Optional Tests |  dupname  asflicense  cc  unit  hbaseprotoc  javac  javadoc  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux a4c400ecf252 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / be432b7c45 |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-352/2/artifact/out/diff-checkstyle-hbase-server.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-352/2/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-352/2/testReport/ |
| Max. process+thread count | 5017 (vs. ulimit of 10000) |
| modules | C: hbase-protocol-shaded hbase-protocol hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-352/2/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 25 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| 0 | mvndep | 14 | Maven dependency ordering for branch |
| +1 | mvninstall | 248 | master passed |
| +1 | compile | 117 | master passed |
| +1 | checkstyle | 88 | master passed |
| +1 | shadedjars | 275 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 434 | master passed |
| +1 | javadoc | 52 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 16 | Maven dependency ordering for patch |
| +1 | mvninstall | 242 | the patch passed |
| +1 | compile | 108 | the patch passed |
| +1 | cc | 108 | the patch passed |
| +1 | javac | 108 | the patch passed |
| +1 | checkstyle | 87 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 265 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 735 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | hbaseprotoc | 107 | the patch passed |
| +1 | findbugs | 464 | the patch passed |
| +1 | javadoc | 58 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 36 | hbase-protocol-shaded in the patch passed. |
| +1 | unit | 25 | hbase-protocol in the patch passed. |
| -1 | unit | 8844 | hbase-server in the patch failed. |
| +1 | asflicense | 90 | The patch does not generate ASF License warnings. |
| | | 12668 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.replication.TestReplicationSyncUpTool |
|   | hadoop.hbase.replication.TestReplicationSmallTests |
|   | hadoop.hbase.replication.multiwal.TestReplicationEndpointWithMultipleWAL |
|   | hadoop.hbase.replication.multiwal.TestReplicationEndpointWithMultipleAsyncWAL |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-352/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/352 |
| Optional Tests |  dupname  asflicense  cc  unit  hbaseprotoc  javac  javadoc  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux fed590497168 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 605f8a15bb |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-352/3/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-352/3/testReport/ |
| Max. process+thread count | 5004 (vs. ulimit of 10000) |
| modules | C: hbase-protocol-shaded hbase-protocol hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-352/3/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by gjacoby126** [comment]
@apurtell @saintstack @Apache9 - adding extended attributes to WALKey as discussed in HBASE-22622. I wasn't able to use a protobuf map class because the "public" protobuf version doesn't support it so it wouldn't be backwards compatible, so I went back to @apurtell 's original suggestion of a repeated key/value attribute. Keys are strings rather than bytes because Java uses reference equality for hashmaps with byte[] keys, and it seemed better than littering the code with ByteBuffers or ImmutableBytesWritable everywhere.  

I've fixed the checkstyle issues, and the test failures appear to be flapping replication tests (each run gives different results even when only trivial formatting changes have been made)

Related coprocessor changes will be in HBASE-22623, unless the community feels it would be better to consolidate the two patches. 

Thanks for taking a look!

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 30 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| 0 | mvndep | 17 | Maven dependency ordering for branch |
| +1 | mvninstall | 263 | master passed |
| +1 | compile | 122 | master passed |
| +1 | checkstyle | 93 | master passed |
| +1 | shadedjars | 285 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 492 | master passed |
| +1 | javadoc | 62 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 14 | Maven dependency ordering for patch |
| +1 | mvninstall | 256 | the patch passed |
| -1 | compile | 45 | hbase-server in the patch failed. |
| -1 | cc | 45 | hbase-server in the patch failed. |
| -1 | javac | 45 | hbase-server in the patch failed. |
| +1 | checkstyle | 94 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 291 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 796 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | hbaseprotoc | 117 | the patch passed |
| +1 | findbugs | 520 | the patch passed |
| +1 | javadoc | 62 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 39 | hbase-protocol-shaded in the patch passed. |
| +1 | unit | 29 | hbase-protocol in the patch passed. |
| +1 | unit | 9069 | hbase-server in the patch passed. |
| +1 | asflicense | 67 | The patch does not generate ASF License warnings. |
| | | 13184 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.7 Server=18.09.7 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-352/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/352 |
| Optional Tests |  dupname  asflicense  cc  unit  hbaseprotoc  javac  javadoc  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 5f07deb6bb9a 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 438bf3284e |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| compile | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-352/4/artifact/out/patch-compile-hbase-server.txt |
| cc | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-352/4/artifact/out/patch-compile-hbase-server.txt |
| javac | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-352/4/artifact/out/patch-compile-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-352/4/testReport/ |
| Max. process+thread count | 4519 (vs. ulimit of 10000) |
| modules | C: hbase-protocol-shaded hbase-protocol hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-352/4/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by gjacoby126** [comment]
@HorizonNet - rebased on master, and was able to build successfully locally. Hopefully now that I've pushed again that will fix it.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 29 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| 0 | mvndep | 12 | Maven dependency ordering for branch |
| +1 | mvninstall | 224 | master passed |
| +1 | compile | 106 | master passed |
| +1 | checkstyle | 81 | master passed |
| +1 | shadedjars | 257 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 371 | master passed |
| +1 | javadoc | 52 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 14 | Maven dependency ordering for patch |
| +1 | mvninstall | 230 | the patch passed |
| +1 | compile | 105 | the patch passed |
| +1 | cc | 105 | the patch passed |
| +1 | javac | 105 | the patch passed |
| +1 | checkstyle | 81 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 252 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 687 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | hbaseprotoc | 104 | the patch passed |
| +1 | findbugs | 450 | the patch passed |
| +1 | javadoc | 52 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 33 | hbase-protocol-shaded in the patch passed. |
| +1 | unit | 23 | hbase-protocol in the patch passed. |
| +1 | unit | 7918 | hbase-server in the patch passed. |
| +1 | asflicense | 72 | The patch does not generate ASF License warnings. |
| | | 11501 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.7 Server=18.09.7 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-352/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/352 |
| Optional Tests |  dupname  asflicense  cc  unit  hbaseprotoc  javac  javadoc  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 6b6ecb9d1bae 4.4.0-139-generic #165-Ubuntu SMP Wed Oct 24 10:58:50 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 74731c2a46 |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-352/5/testReport/ |
| Max. process+thread count | 4415 (vs. ulimit of 10000) |
| modules | C: hbase-protocol-shaded hbase-protocol hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-352/5/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/wal/WALKey.java [hunk 1]
```diff
@@ -86,6 +86,21 @@ default long getNonce() {
...
+     * @param attributeKey The key of a key / value pair
+     */
+  default byte[] getExtendedAttribute(String attributeKey){
+    return null;
+  }
+
+    /**
+     * Returns a map of all extended attributes injected into this WAL key.
+     */
+  default Map<String, byte[]> getExtendedAttributes() {
```
Comment (apurtell) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/wal/WALKey.java]: Missing setters?

Comment (gjacoby126) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/wal/WALKey.java]: @apurtell - WALKey's interface comments specifically say that setters aren't permitted. They're meant to be immutable, so the extended attributes will be set during construction of the WALKeyImpl

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 52 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| 0 | mvndep | 14 | Maven dependency ordering for branch |
| +1 | mvninstall | 308 | master passed |
| +1 | compile | 146 | master passed |
| +1 | checkstyle | 110 | master passed |
| +1 | shadedjars | 359 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 575 | master passed |
| +1 | javadoc | 75 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 16 | Maven dependency ordering for patch |
| +1 | mvninstall | 316 | the patch passed |
| +1 | compile | 144 | the patch passed |
| +1 | cc | 144 | the patch passed |
| +1 | javac | 144 | the patch passed |
| +1 | checkstyle | 107 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 347 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 970 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | hbaseprotoc | 146 | the patch passed |
| +1 | findbugs | 606 | the patch passed |
| +1 | javadoc | 70 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 47 | hbase-protocol-shaded in the patch passed. |
| +1 | unit | 31 | hbase-protocol in the patch passed. |
| -1 | unit | 15810 | hbase-server in the patch failed. |
| +1 | asflicense | 85 | The patch does not generate ASF License warnings. |
| | | 20772 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.client.TestSnapshotTemporaryDirectory |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.7 Server=18.09.7 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-352/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/352 |
| Optional Tests |  dupname  asflicense  cc  unit  hbaseprotoc  javac  javadoc  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 1d679ecf22ae 4.4.0-154-generic #181-Ubuntu SMP Tue Jun 25 05:29:03 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / cc38de1a39 |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-352/6/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-352/6/testReport/ |
| Max. process+thread count | 4542 (vs. ulimit of 10000) |
| modules | C: hbase-protocol-shaded hbase-protocol hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-352/6/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 55 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| 0 | mvndep | 26 | Maven dependency ordering for branch |
| +1 | mvninstall | 246 | master passed |
| +1 | compile | 109 | master passed |
| +1 | checkstyle | 85 | master passed |
| +1 | shadedjars | 264 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 444 | master passed |
| +1 | javadoc | 58 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 14 | Maven dependency ordering for patch |
| +1 | mvninstall | 240 | the patch passed |
| +1 | compile | 107 | the patch passed |
| +1 | cc | 107 | the patch passed |
| +1 | javac | 107 | the patch passed |
| +1 | checkstyle | 86 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 265 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 733 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| +1 | hbaseprotoc | 108 | the patch passed |
| +1 | findbugs | 462 | the patch passed |
| +1 | javadoc | 55 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 35 | hbase-protocol-shaded in the patch passed. |
| +1 | unit | 23 | hbase-protocol in the patch passed. |
| -1 | unit | 14459 | hbase-server in the patch failed. |
| +1 | asflicense | 77 | The patch does not generate ASF License warnings. |
| | | 18287 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.7 Server=18.09.7 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-352/7/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/352 |
| Optional Tests |  dupname  asflicense  cc  unit  hbaseprotoc  javac  javadoc  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux b0efca1f5f12 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 72e58a8fed |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-352/7/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-352/7/testReport/ |
| Max. process+thread count | 5029 (vs. ulimit of 10000) |
| modules | C: hbase-protocol-shaded hbase-protocol hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-352/7/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by apurtell** [comment]
Test failures in latest precommit look environmental in nature. Will check locally before merging this.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/wal/WALKeyImpl.java [hunk 2]
```diff
@@ -573,6 +622,12 @@ public void readFieldsFromPb(WALProtos.WALKey walKey,
     if (walKey.hasOrigSequenceNumber()) {
       this.origLogSeqNum = walKey.getOrigSequenceNumber();
     }
+    if (walKey.getExtendedAttributesCount() > 0){
+      this.extendedAttributes = new HashMap<>(walKey.getExtendedAttributesCount());
+      for (WALProtos.Attribute attr : walKey.getExtendedAttributesList()){
+        extendedAttributes.put(attr.getKey(), attr.getValue().toByteArray());
```
Comment (apurtell) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/wal/WALKeyImpl.java]: Noticed while backporting. Attribute values may have been compressed and need to be decompressed. Will fix up for commit.

**Pull request closed without merging**